src/coreclr/jit/earlyprop.cpp

```diff
@@ -457,6 +457,10 @@ bool Compiler::optFoldNullCheck(GenTree* tree, LocalNumberToNullCheckTreeMap* nu
         nullCheckTree->SetHasOrderingSideEffect();
         nullCheckTree->gtFlags |= GTF_IND_NONFAULTING;
 
+        // The current indir is no longer non-faulting.
+        tree->gtFlags &= ~GTF_IND_NONFAULTING;
+        tree->SetIndirExceptionFlags(this);
```

Comment:
I presume in theory we can avoid removing GTF_IND_NONFAULTING and adding GTF_EXCEPT if the leading (if it is in fact leading) nullcheck had them, but wanted to be on a safe side

Reply:
IS the `SetIndirExceptionFlags` necessary? I assume it should be recomputed below when all side effect flags are updated.

Reply:
Correct, GTF_EXCEPT is set later via gtUpdateNodeOperSideEffects, so this one is redundant